src/box/lua/config/instance_config.lua

```diff
@@ -1694,7 +1707,108 @@ return schema.new('instance_config', schema.record({
+            apply_default_if = audit_apply_default_if,
+            validate = audit_validate,
+        }),
+        file = schema.scalar({
+            type = 'string',
+            default = 'var/log/{{ instance_name }}/audit.log',
+            apply_default_if = audit_apply_default_if,
+            validate = audit_validate,
+        }),
+        pipe = schema.scalar({
+            type = 'string',
+            default = box.NULL,
+            apply_default_if = audit_apply_default_if,
+            validate = audit_validate,
+        }),
+        syslog = schema.record({
+            identity = schema.scalar({
+                type = 'string',
+                default = 'tarantool',
+                apply_default_if = audit_apply_default_if,
+                validate = audit_validate,
+            }),
+            facility = schema.scalar({
+                type = 'string',
+                default = 'local7',
+                apply_default_if = audit_apply_default_if,
+                validate = audit_validate,
+            }),
+            server = schema.scalar({
+                type = 'string',
+                -- The logger tries /dev/log and then
+                -- /var/run/syslog if no server is provided.
+                default = box.NULL,
+                apply_default_if = audit_apply_default_if,
+                validate = audit_validate,
+            }),
+        }),
+        nonblock = schema.scalar({
+            type = 'boolean',
+            box_cfg = 'audit_nonblock',
+            box_cfg_nondynamic = true,
+            default = false,
+            apply_default_if = audit_apply_default_if,
+            validate = audit_validate,
+        }),
+        format = schema.enum({
+            'plain',
+            'json',
+            'csv',
+        }, {
+            box_cfg = 'audit_format',
+            default = 'json',
+            apply_default_if = audit_apply_default_if,
+            validate = audit_validate,
+        }),
+        filter = schema.set({
+            "audit_enable",
+            "custom",
+            "auth_ok",
+            "auth_fail",
+            "disconnect",
+            "user_create",
+            "user_drop",
+            "role_create",
+            "role_drop",
+            "user_enable",
+            "user_disable",
+            "user_grant_rights",
+            "user_revoke_rights",
+            "role_grant_rights",
+            "role_revoke_rights",
+            "password_change",
+            "access_denied",
+            "eval",
+            "call",
+            "space_select",
+            "space_create",
+            "space_alter",
+            "space_drop",
+            "space_insert",
+            "space_replace",
+            "space_delete",
+            "none",
+            "all",
+            "audit",
+            "auth",
+            "priv",
+            "ddl",
+            "dml",
+            "data_operations",
+            "compatibility",
+        }),
```

Comment:
1. Can we write a test that all the events and events group are covered?
2. I would like to have some explicit separation of groups and events.

Reply:
3. This option should have a validate in a future, so it would be good to add a TODO comment similar to `sharding.roles` one.

Reply:
1. Done.
2. Added a comment. Should I make different options?
3. Done.

Reply:
1. I meant that it should be good to check a consistency between available `box.cfg.audit_filter` options and instance_config's ones.
2. OK. No, that's is exactly what I asked.
3. OK.

Thanks!